.github/workflows/golangci-lint.yaml

```diff
@@ -0,0 +1,52 @@
+on:
```

Comment:
Why did you move this out into a separate workflow? My only concern is the extra time related to installing dependencies and such.

Reply:
@Olshansk Reading through the link @okdas posted in issue #295 I saw this https://github.com/golangci/golangci-lint-action/issues/23#issuecomment-647177770 which says to separate the workflow into its own job - i went one further into its own file - I can put it back in but it needs its own job so that the errors like the ones below don't happen anymore

```bash
Cannot open: File exists
```

Reply:
Makes sense. Can you just add a comment above the `job` field explaining why this is a separate job in its own file? You can probably repurpose most of your comment above.

Reply:
Saw your comment. Ignore this & resolve.

Reply:
@Olshansk I left a new comment below this was the intention however I double checked and this didn't actually solve those errors they still occurred. I have since moved the job back into the test job only difference now is it fails on linting errors and looks for new errors in the PR. I will keep looking into how to solve these file exists errors